### ISSUE TITLE
"add_offset" and "scale_factor" is not applied correctly in multi-file .nc datasets

### ISSUE THREAD

**Issue by Datseris** [Alexander-Barth/NCDatasets.jl#109]
Hi,
I am wondering whether these two factors are applied properly (and automatically) in multi-file loading. I had a problem where I downloaded data and once I animated the temporal evolution of the spatio temporal field, at each part of the animation from December to January the field values changed dramatically. 

Each year of my data is in a different file, and it turns out that each file has significantly different values for the offset and scale factor. Here is an example:

```julia
path = readdir(datadir("ERA5"); join = true)[1:end-1] # last year is problematic, it has one extra dimension

ncdata = NCDataset(path; aggdim = "time") # animating time evolution of this data has discontinuities

fieldname = "vertical_velocity"
accessname = "w"

for p in path
    ncdata = NCDataset(p)
    X = ncdata[accessname]
    @show X.attrib["add_offset"], X.attrib["scale_factor"]
end
```

which outputs

```
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6380269135861719, 7.923328297848344e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.580450491903175, 7.620096614137319e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6749206420174069, 7.898879233479578e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6895523444253533, 7.532528274679564e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.74354152857858, 8.239389264454252e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6069337565824714, 8.300162845330009e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.7633918383374232, 8.991188716523158e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6961179462645649, 7.705140626445246e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6745299908489549, 7.732451546898516e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6423474029119253, 7.91995892048777e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.6028299629265329, 9.149321431710395e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.5241055198044138, 8.874981988328221e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.35945312282096925, 8.378226257204771e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.7059709446794712, 7.18201861930818e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.4734058222959074, 9.158416795080818e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.5730201126558377, 7.817716940350879e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.7246898602779371, 7.935537186007729e-5)
(X.attrib["add_offset"], X.attrib["scale_factor"]) = (0.5035102036511804, 8.065165760021132e-5)
```

a co-worker loaded exactly the same data using xarray and he did not have these discontinuities. We both now believe that the problem is that the application of the scale factors is not done correctly here for multi file datasets. The main reason I am believing this, is because I do identical processing in another field (specifically cloud coverage), where the offset factors are almost identical in each year. There I don't see any discontinuities.

**Comment by Datseris** [comment]
I've just loaded exactly the same data using Python's xarray, which I then transformed to standard Julia array and I did exactly the same pipeline of operations as when I loaded the data with NCDatasets.jl

There were no discontinuities or anything weird happening in this case. I think this is a bug in NCDatasets.jl where these factors are not applied correctly in multi file data. I'm happy to share my data for testing, they are about 250 mb.
